src/components/data_search/DatasetSearchTable.js

```diff
@@ -102,6 +105,13 @@ export const DatasetSearchTable = (props) => {
     history.push(`/dar_application/${darDraft.referenceId}`);
   };
 
+  const handleSearchChange = useCallback((searchTerms) => searchOnFilteredList(
+    searchTerms,
+    datasets,
+    getSearchFilterFunctions().datasetTerms,
+    setFiltered
+  ), [datasets]);
```

Comment:
I think `getSearchFilterFunctions().datasetTerms` needs to be integrated with the `filterHandler` function instead of being handled separately here. This way, the same function that implements the faceted search also implements the free text filter.